cpu/native/Kconfig

```diff
@@ -38,10 +38,10 @@ config CPU_MODEL_NATIVE
     select NATIVE_OS_FREEBSD if "$(OS)" = "FreeBSD"
 
 ## Declaration of specific features
-config HAS_ARCH_NATIVE
+config HAS_ARCH_X86_32
```

Comment:
I can still compile RIOT native on a Raspberry Pi, so this is a lie.

Reply:
we really need to fix that
it `HAS_ARCH_ARMv6` in that case? or do you compile AARCH64 (i tried and failed)
how to detect?

Reply:
I thought `native` would be entirely arch independent, but there is indeed some code in `cpu/native` that implements special handling `__arm__`.

Anyway, does that matter here? Wouldn't that just be `HAS_ARCH_NATIVE_32`?

Reply:
I think it does if one wants to use this information without guessing in every module/pkg that requires this information. ATM it seems as if native is, be arch independent but it is not.

Reply:
@benpicco @kfessel how do we move forward here? I think `HAS_ARCH_NATIVE_32` makes sense.

Reply:
HAS_ARCH_NATIVE_32 makes sense (good extra attribute), but it is is not enough, since pkg might have own "*make chains" that need to select compiler or assembler or code according to the real architecture. Esspecial if we consider native archs aside from x86, aarch64 is comming in huge steps (and there might be a RISC-V with an OS (Linux) on the horizon).